src/modules/sync/sync.ts

```diff
@@ -1,29 +1,25 @@
-import {
-  ContentTypeElements,
-  ContentTypeModels,
-  ContentTypeSnippetModels,
-  ManagementClient,
-  SharedContracts,
-  TaxonomyModels,
-} from "@kontent-ai/management-sdk";
-import { match, P } from "ts-pattern";
+import { ManagementClient } from "@kontent-ai/management-sdk";
 
 import { logInfo, LogOptions } from "../../log.js";
-import { omit } from "../../utils/object.js";
 import { serially } from "../../utils/requests.js";
-import { notNullOrUndefined } from "../../utils/typeguards.js";
-import { RequiredCodename } from "../../utils/types.js";
-import { elementTypes } from "./constants/elements.js";
-import { ElementsTypes } from "./types/contractModels.js";
-import { DiffModel, WebSpotlightDiffModel } from "./types/diffModel.js";
-import { getTargetCodename, PatchOperation } from "./types/patchOperation.js";
-
-const referencingElements: ReadonlyArray<ElementsTypes> = ["rich_text", "modular_content", "subpages"];
-const referenceProps = ["allowed_content_types", "allowed_item_link_types"] as const;
-type PropName = typeof referenceProps[number];
+import { syncAssetFolders } from "./sync/assetFolders.js";
+import {
+  addElementsIntoSnippetsWithoutReferences,
+  addSnippetsReferences,
+  addSnippetsWithoutReferences,
+  deleteSnippet,
+  updateSnippets,
+} from "./sync/snippets.js";
+import { syncTaxonomies } from "./sync/taxonomy.js";
+import { addTypesWithoutReferences, deleteContentType, updateContentTypesAndAddReferences } from "./sync/types.js";
+import { isOp } from "./sync/utils.js";
+import { updateWebSpotlight } from "./sync/webSpotlight.js";
+import { DiffModel } from "./types/diffModel.js";
 
 export const sync = async (client: ManagementClient, diff: DiffModel, logOptions: LogOptions) => {
   // there order of these operations is very important
+  await syncAssetFolders(client, diff.assetFolders, logOptions);
```

Comment:
The only options to validate this are to:
* Load all the assets in the project (this is too expensive for bigger projects)
* Delete the folder and see if there is an error. Then re-create it if it doesn't fail. (re-creating the folder doesn't create it in the original place so this validation is not without side effects) Also if the user kills the command before the validation ends they can end up with some folders deleted without actually running anything that could be destructive.

I mentioned the limitation in the readme file to at least make it visible there.

Reply:
I can split the remove operations into separate requests and catch those that fail. This will at least prevent interrupting the command and will print a helpful message during the command run.